File: app/src/main/java/com/example/app/room/Room.java
package com.example.app.room;

import com.example.app.ongoingquiz.OngoingQuiz;
import lombok.Builder;

import java.util.Set;

public record Room(String code, OngoingQuiz ongoingQuiz, String moderator, Set<String> players) {
    @Builder
    public Room {
    }
}

File: frontend/src/domain/Room.ts
export interface CurrentQuestion {
  text: string;
  answers: string[];
}

export enum OngoingQuizStatus {
  NOT_STARTED = 'NOT_STARTED',
  QUESTION_IN_PROGRESS = 'QUESTION_IN_PROGRESS',
  WAITING = 'WAITING',
  DONE = 'DONE'
}

export interface OngoingQuiz {
  currentQuestion: CurrentQuestion | undefined;
  points: { [playerName: string]: number },
  status: OngoingQuizStatus
}

export interface Room {
  code: string;
  ongoingQuiz?: OngoingQuiz;
  moderator: string;
  players: string[];
}